router.go

```diff
@@ -63,39 +72,76 @@ func (m MapRouter) Register(rs []transport.Procedure) {
 			panic("Expected procedure name not to be empty string in registration")
 		}
 
+		if r.Encoding != "" {
```

Comment:
oh, i was thinking of having this function in tandem with the "Procedures" function below, I see how it got misconstrued with actually routing to procedures (which wouldn't work for the reasons you specified).